Wrappers/Python/cil/optimisation/algorithms/Algorithm.py

```diff
@@ -26,31 +26,22 @@
 
 
 class Algorithm:
-    '''Base class for iterative algorithms
+    r"""Base class providing minimal infrastructure for iterative algorithms.
 
-      provides the minimal infrastructure.
+   An iterative algorithm is designed to solve an optimization problem by repeatedly refining a solution. In CIL, we use iterative algorithms to minimize an objective function, often referred to as a loss. The process begins with an initial guess, and with each iteration, the algorithm updates the current solution based on the results of previous iterations (previous iterates). Iterative algorithms typically continue until a stopping criterion is met, indicating that an optimal or sufficiently good solution has been found. In CIL, stopping criteria can be implemented using a callback function (`cil.optimisation.utilities.callbacks`).
+     
+    The user is required to implement the :code:`set_up`, :code:`__init__`, :code:`update` and :code:`update_objective` methods.
 
-      Algorithms are iterables so can be easily run in a for loop. They will
-      stop as soon as the stop criterion is met.
-      The user is required to implement the :code:`set_up`, :code:`__init__`, :code:`update` and
-      and :code:`update_objective` methods
-
-      A courtesy method :code:`run` is available to run :code:`n` iterations. The method accepts
-      a :code:`callbacks` list of callables, each of which receive the current Algorithm object
-      (which in turn contains the iteration number and the actual objective value)
-      and can be used to trigger print to screens and other user interactions. The :code:`run`
-      method will stop when the stopping criterion is met or `StopIteration` is raised.
-   '''
+    The method :code:`run` is available to run :code:`n` iterations. The method accepts a :code:`callbacks` list of callables, each of which receive the current Algorithm object (which in turn contains the iteration number and the actual objective value) and can be used to trigger print to screens and other user interactions. The :code:`run` method will stop when the stopping criterion is met or `StopIteration` is raised.
+    
+    Parameters
+    ----------
+    update_objective_interval: int, optional, default 1
+        the interval every which we would save the current objective. 1 means every iteration, 2 every 2 iteration and so forth. This is by default 1 and should be increased when evaluating the objective is computationally expensive.
```

Comment:
```suggestion
        the interval in which we would save the current objective. 1 means every iteration, 2 every 2 iteration and so forth. This is by default 1 and should be increased when evaluating the objective is computationally expensive.
```

Reply:
Feel like it should be the interval in or the interval after?

Reply:
Good spot Laura - I have tried to make this clearer